lib/src/git_subprocess.rs

```diff
@@ -33,6 +33,8 @@ use crate::git::RefToPush;
 use crate::git::RemoteCallbacks;
 use crate::git_backend::GitBackend;
 
+const MINIMUM_GIT_VERSION: &str = "2.40.0";
```

Comment:
Can you copy the comment why we chose 2.40.0? It will help when we have to update the minimum version.
https://github.com/jj-vcs/jj/pull/5979#discussion_r1990504182